### PR TITLE
Jenkinsfile: limit to running on nodes labeled as proofs

### PR DESCRIPTION
**Pull request by ehildenb**
This should keep a single instance of this build running at a time on nodes with lots of memory.

### CODE REVIEW

**Comment by denis-bogdanas** [comment]
jenkins: test this please

File: resources/kprove.mak [hunk 1]
```diff
@@ -78,6 +78,7 @@ $(K_REPO_DIR):
 	git clone $(K_REPO_URL) $(K_REPO_DIR)
 	cd $(K_REPO_DIR) \
 		&& git reset --hard $(K_VERSION) \
+		&& git submodule update --init --recursive \
```
Comment (daejunpark) [inline resources/kprove.mak]: @denis-bogdanas why is this needed? we will not need llvm nor haskell backend, right?

Comment (denis-bogdanas) [reply resources/kprove.mak]: ^^^ @ehildenb
